src/Altinn.Platform/Altinn.Platform.Storage/UnitTest/TestingControllers/InstancesControllerTests.cs

```diff
@@ -276,6 +276,63 @@ public async void Delete_ReponseIsDeny_ReturnsStatusForbidden()
                 Assert.Equal(HttpStatusCode.Forbidden, response.StatusCode);
             }
 
+            /// <summary>
+            /// Test case: App owner tries to hard delete an instance
+            /// Expected: Returns success and deleted instance
+            /// </summary>
+            [Fact]
+            public async void Delete_OrgHardDeletesInstance_DeletedInstanceIsReturned()
+            {
+                // Arrange
+                int instanceOwnerId = 1337;
+                string instanceGuid = "7e6cc8e2-6cd4-4ad4-9ce8-c37a767677b5";
+
+                string requestUri = $"{BasePath}/{instanceOwnerId}/{instanceGuid}?hard=true";
+
+                HttpClient client = GetTestClient();
+                string token = PrincipalUtil.GetOrgToken("tdd");
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+                // Act
+                HttpResponseMessage response = await client.DeleteAsync(requestUri);
+
+                string json = await response.Content.ReadAsStringAsync();
+                Instance deletedInstance = JsonConvert.DeserializeObject<Instance>(json);
+
+                // Assert
+                Assert.NotNull(deletedInstance.Status.HardDeleted);
+                Assert.NotNull(deletedInstance.Status.SoftDeleted);
+                Assert.Equal(deletedInstance.Status.HardDeleted, deletedInstance.Status.SoftDeleted);
+            }
+
+            /// <summary>
+            /// Test case: End user system tries to soft delete an instance
+            /// Expected: Returns success and deleted instance
+            /// </summary>
+            [Fact]
+            public async void Delete_EndUserSoftDeletesInstance_DeletedInstanceIsReturned()
```

Comment:
DeletedInstanceIsReturned could be more specific. For example: ReturnedInstanceHasStatusOnlySoftDeleted.